cmd/sonobuoy/app/version.go

```diff
@@ -20,20 +20,76 @@ import (
 	"fmt"
 
 	"github.com/heptio/sonobuoy/pkg/buildinfo"
+	"github.com/heptio/sonobuoy/pkg/errlog"
+	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
 )
 
-func init() {
-	RootCmd.AddCommand(versionCmd)
+type versionFlags struct {
+	kubecfg Kubeconfig
 }
 
-var versionCmd = &cobra.Command{
-	Use:   "version",
-	Short: "Print sonobuoy version",
-	Run:   runVersion,
-	Args:  cobra.ExactArgs(0),
+var versionflags versionFlags
+
+func init() {
+	cmd := &cobra.Command{
+		Use:   "version",
+		Short: "Print sonobuoy version",
+		Run:   runVersion,
+		Args:  cobra.ExactArgs(0),
+	}
+
+	AddKubeconfigFlag(&versionflags.kubecfg, cmd.Flags())
+
+	RootCmd.AddCommand(cmd)
 }
 
 func runVersion(cmd *cobra.Command, args []string) {
-	fmt.Println(buildinfo.Version)
+
+	fmt.Println(fmt.Sprintf("Sonobuoy Version: %s", buildinfo.Version))
+	fmt.Println(fmt.Sprintf("MinimumKubeVersion: %s", buildinfo.MinimumKubeVersion))
+	fmt.Println(fmt.Sprintf("MaximumKubeVersion: %s", buildinfo.MaximumKubeVersion))
+	fmt.Println(fmt.Sprintf("GitSHA: %s", buildinfo.GitSHA))
+
+	// Get Kubernetes version, this is last so that the regular version information
+	// will be shown even if the API server cannot be contacted and throws an error
+	apiVersion, skipk8sCheck := getK8Sversion()
+	if !skipk8sCheck {
+		fmt.Println("API Version: ", apiVersion)
+	} else {
+		fmt.Println("API Version check skipped due to missing `--kubeconfig` or other error")
+	}
+}
+
+func getK8Sversion() (string, bool) {
```

Comment:
`(string,err)` would be a more clear signature. Then the caller can decide to expose the error or not.

Reply:
Yeah 100%, I refactored this out of the other method and didn't come back and update this.